src/opensearch_sdk_py/actions/request_handler.py

```diff
@@ -8,29 +8,31 @@
 #
 
 import logging
-from abc import ABC, abstractmethod
+from abc import abstractmethod
 from typing import Optional
 
+from opensearch_sdk_py.actions.request_response_handler import RequestResponseHandler
 from opensearch_sdk_py.extension import Extension
-from opensearch_sdk_py.transport.outbound_message import OutboundMessage
 from opensearch_sdk_py.transport.outbound_message_request import OutboundMessageRequest
+from opensearch_sdk_py.transport.outbound_message_response import OutboundMessageResponse
 from opensearch_sdk_py.transport.stream_input import StreamInput
 from opensearch_sdk_py.transport.stream_output import StreamOutput
 
 
-class RequestHandler(ABC):
+class RequestHandler(RequestResponseHandler):
     def __init__(self, action: str, extension: Extension):
         self.action = action
         self.extension = extension
+        self.response: OutboundMessageResponse = None
 
     @abstractmethod
     def handle(self, request: OutboundMessageRequest, input: StreamInput = None) -> Optional[bytes]:
         pass  # pragma: no cover
 
-    def send(self, message: OutboundMessage) -> StreamOutput:
+    def send(self) -> StreamOutput:
         output = StreamOutput()
-        message.write_to(output)
+        self.response.write_to(output)
         raw_out = output.getvalue()
-        logging.info(f"> {message.__str__()}, size={len(raw_out)} byte(s)")
+        logging.info(f"> {self.response.__str__()}, size={len(raw_out)} byte(s)")
```

Comment:
Shouldn't need `__str__()` here, I think it's implied by `{self.response}`, or maybe use `str(self.response)`.